packages/mui-joy/src/Typography/Typography.test.tsx

```diff
@@ -40,11 +40,13 @@ describe('<Typography />', () => {
 
   ['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'body1', 'body2', 'body3'].forEach((level) => {
     it(`should render ${level} text`, () => {
-      const { container } = render(<Typography level={level}>Hello</Typography>);
+      const { container } = render(
+        <Typography level={level as TypographyProps['level']}>Hello</Typography>,
+      );
 
       expect(classes).to.have.property(level);
-
-      expect(container.firstChild).to.have.class(classes[level]);
+      const classToCheck: string | RegExp = classes[level as keyof typeof classes];
+      expect(container.firstChild).to.have.class(classToCheck);
```

Comment:
Please revert the change. No error should be thrown if you address my comment above.